src/components/DatePicker/DatePicker.jsx

```diff
@@ -67,6 +66,14 @@ export const DatePicker = ({
         onMonthChange?.(currentMonth);
     }, [currentMonth, onMonthChange]);
 
+    useEffect(() => {
+        if (timezoneName) {
+            dayjs.tz.setDefault(timezoneName);
+        } else {
+            dayjs.tz.setDefault();
+        }
+    }, [timezoneName]);
```

Comment:
What will happen if there is already a date in the state, and then timezone changes? Suddenly your internal dates are not in the same timezone as your default timezone. This can cause a lot of confusion.

I think when timezone changes, either we update the internal dates, or we reset the entire state.